packages/plugin-core/src/commands/NoteLookupCommand.ts

```diff
@@ -506,6 +510,51 @@ export class NoteLookupCommand
     return { uri, node: nodeNew, resp };
   }
 
+  /**
+   * Find corresponding note for given schema template. Supports cross-vault lookup.
+   *
+   * @param schemaTemplate
+   */
+  private async getNoteForSchemaTemplate(schemaTemplate: SchemaTemplate) {
+    let maybeVault: DVault | undefined;
+    const { link, vaultName } = this.parseDendronURI(schemaTemplate.id);
+    const engine = ExtensionProvider.getEngine();
+
+    // If vault is specified, lookup by template id + vault
+    if (!_.isUndefined(vaultName)) {
+      maybeVault = VaultUtils.getVaultByNameOrThrow({
+        vname: vaultName,
+        vaults: engine.vaults,
+      });
+    }
+
+    const maybeNote = await WSUtilsV2.instance().getNoteFromMultiVault({
+      fname: link,
+      vault: maybeVault,
+    });
+    if (_.isUndefined(maybeNote)) {
+      throw new DendronError({
+        message: `No template found for ${schemaTemplate.id}`,
+      });
+    }
+    return maybeNote;
+  }
+
+  private parseDendronURI(linkString: string) {
```

Comment:
We already have `parseRef`, could we use that here instead? https://github.com/dendronhq/dendron/blob/68185d0895df3e0514d2ec88c7b778c5f24de18d/packages/plugin-core/src/utils/md.ts#L389

Reply:
Oh nice, thanks for letting me know